### PR TITLE
feat: Consume enroll verb builder in AtLookup.executeVerb()

### LINKED ISSUE #655: Missing implementation for enroll verb handler in AtLookup.executeVerb
### Describe the bug

When AtLookup.executeVerb() is called with EnrollVerbBuilder, it does not have the required calls to execute the EnrollVerbBuilder. It throws a 

### Expected behavior

When EnrollVerbBuilder is passed to AtLookup.executeVerb(), it should build the command, send it to the server and then read the response.

### PR DESCRIPTION
**Pull request by srieteja**
Closes https://github.com/atsign-foundation/at_libraries/issues/655

**- What I did**
- add missing impl for EnrollVerbHandler in AtLookup.executeVerb()
- upgrade at_commons to v4.1.1 and at_Utils to v3.0.18

**- How to verify it**

**- Description for the changelog**

- add missing impl for EnrollVerbHandler in AtLookup.executeVerb()

### CODE REVIEW

File: packages/at_lookup/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+## 3.0.48
+- feat: consume EnrollVerbBuilder in AtLookup.executeVerb()
+- chore: upgrade at_commons to v4.1.1 and at_Utils to v3.0.18
```
Comment (murali-shris) [inline packages/at_lookup/CHANGELOG.md]: change the case at_Utils --> at_utils

File: packages/at_lookup/lib/src/at_lookup_impl.dart [hunk 2]
```diff
@@ -407,6 +410,11 @@ class AtLookupImpl implements AtLookUp {
     return await _process(atCommand, auth: true);
   }
 
+  Future<String> _enroll(EnrollVerbBuilder builder) async {
```
Comment (murali-shris) [inline packages/at_lookup/lib/src/at_lookup_impl.dart:413]: need unit or functional test to verify this method

File: packages/at_lookup/lib/src/at_lookup_impl.dart [hunk 3]
```diff
@@ -407,6 +410,11 @@ class AtLookupImpl implements AtLookUp {
...
+    var atCommand = builder.buildCommand();
+    return await _process(atCommand, auth: true);
```
Comment (murali-shris) [inline packages/at_lookup/lib/src/at_lookup_impl.dart:418]: enroll is not an authenticated verb is all cases

**Review by murali-shris — CHANGES_REQUESTED** [review]

File: packages/at_lookup/CHANGELOG.md (hunk 1 above)
Comment (srieteja) [reply packages/at_lookup/CHANGELOG.md]: change made in [a293b1f](https://github.com/atsign-foundation/at_libraries/pull/657/commits/a293b1f2bb64475a43ed1bce3598f92ce828420c)

File: packages/at_lookup/lib/src/at_lookup_impl.dart (hunk 3 above)
Comment (srieteja) [reply packages/at_lookup/lib/src/at_lookup_impl.dart:418]: change made: send command with auth = false when EnrollVerbBuilder.operation is 'request'. [cdcea33](https://github.com/atsign-foundation/at_libraries/pull/657/commits/cdcea33b7a56f6a1c260ad9e014c7577e5ee37a3)

File: packages/at_lookup/lib/src/at_lookup_impl.dart (hunk 2 above)
Comment (srieteja) [reply packages/at_lookup/lib/src/at_lookup_impl.dart:413]: Unit tests added in [bf70ac7](https://github.com/atsign-foundation/at_libraries/pull/657/commits/bf70ac75fa3b22cd2f9dc2ac0e8a64cfbd63a9bb)

File: packages/at_lookup/lib/src/at_lookup_impl.dart [hunk 4]
```diff
@@ -407,6 +410,14 @@ class AtLookupImpl implements AtLookUp {
     return await _process(atCommand, auth: true);
   }
 
+  Future<String> _enroll(EnrollVerbBuilder builder) async {
+    var atCommand = builder.buildCommand();
+    if(builder.operation == EnrollOperationEnum.request){
```
Comment (murali-shris) [inline packages/at_lookup/lib/src/at_lookup_impl.dart]: please check dart format

**Review by murali-shris — CHANGES_REQUESTED** [review]

File: packages/at_lookup/test/at_lookup_test.dart [hunk 5]
```diff
@@ -306,4 +312,149 @@ void main() {
...
+
+      when(() => mockOutBoundConnection.write(enrollCommand))
+          .thenAnswer((invocation) {
+        mockSecureSocket.write(enrollCommand);
+        return Future.value();
+      });
+      when(() => mockOutboundListener.read())
+          .thenAnswer((_) => Future.value(enrollResponse));
+      AtConnectionMetaData? atConnectionMetaData = OutboundConnectionMetadata()
+        ..isAuthenticated = false;
```
Comment (srieteja) [inline packages/at_lookup/test/at_lookup_test.dart:346]: set 'isAuthenticated' to false for enroll:request test to ensure 'auth : false' is working

File: packages/at_lookup/lib/src/at_lookup_impl.dart (hunk 4 above)
Comment (srieteja) [reply packages/at_lookup/lib/src/at_lookup_impl.dart]: ran dart format for at_lookup in the latest commit

**Review by murali-shris — APPROVED** [review]

**Pull request merged**